infra/proposal.go

```diff
@@ -92,30 +92,23 @@ func CreateSignedTx(proposal *peer.Proposal, signer *Crypto, resps ...*peer.Prop
 		return nil, err
 	}
 
+	endorsements := make([]*peer.Endorsement, 0)
+
 	// ensure that all actions are bitwise equal and that they are successful
 	var a1 []byte
-	for n, r := range resps {
-		if n == 0 {
-			a1 = r.Payload
-			if r.Response.Status < 200 || r.Response.Status >= 400 {
-				return nil, errors.Errorf("proposal response was not successful, error code %d, msg %s", r.Response.Status, r.Response.Message)
-			}
-			continue
+	for r := range resps {
+		a1 = r.Payload
+		if r.Response.Status < 200 || r.Response.Status >= 400 {
+			return nil, errors.Errorf("proposal response was not successful, error code %d, msg %s", r.Response.Status, r.Response.Message)
 		}
-
 		if bytes.Compare(a1, r.Payload) != 0 {
 			return nil, errors.New("ProposalResponsePayloads do not match")
+		} else {
+			endorsements = append(endorsements, r.Endorsement)
```

Comment:
you can leave this outside of `else`:

```
if bytes.Compare(a1, r.Payload) != 0 {
  return nil, errors.New("ProposalResponsePayloads do not match")
}

endorsements = append(endorsements, r.Endorsement)
```

Reply:
in general with Go, we try to use less `else`, so you'll see lots of patterns like:
```
if condition {
  return error
}

do
```
instead of
```
if condition {
  return error
} else {
  do
}
```

Reply:
yep, it is a coding bug.